Adding two staging tests to ci.yaml

yusuf-goog opened the pull request:
Adding a properties block (linux_stating_build) and 2 builds
that use this to ci.yaml. This is to verify some failures on
these builds that are happening when running via entries
in devicelab_stating_config.star file.

Bug:102406

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

yusuf-goog committed:
Adding two staging tests to ci.yaml

Adding a properties block (linux_stating_build) and 2 builds
that use this to ci.yaml. This is to verify some failures on
these builds that are happening when running via entries
in devicelab_stating_config.star file.

Bug:102406

keyonghan commented on .ci.yaml:
```diff
@@ -10,6 +10,23 @@ enabled_branches:
   - flutter-\d+\.\d+-candidate\.\d+
 
 platform_properties:
+  linux_staging_build:
```
I am a bit reluctant to add this set of builders in the .ci.yaml.
Instead, after taking a look at the failure: https://ci.chromium.org/ui/p/flutter/builders/staging/Linux_staging%20analyze/62/overview
you are missing the `validation` property: https://github.com/flutter/flutter/blob/master/.ci.yaml#L222-L223

yusuf-goog replied:
Shouldn't we be moving to this scheme and deprecating running staging tests with devicelab_staging_config.star, to avoid having to maintain 2 different methods of running staging tests, and avoid issues like this where config is different?

If we choose to keep devicelab_staging_config.star as a way of testing staging tests, we will continue to experience inconsistencies and overhead with our main ci.yaml based mechanism of defining builds.

CaseyHillers replied:
+1 to moving it into the ci.yaml. I think some ways to simplify it is we could update the ci.yaml infra to instead not require an explicit staging platform_properties, but other than that this approach LGTM

keyonghan replied:
How do we validate all Linux builders (debian->ubuntu) work without affecting our end users? With this, we need to add ~100 new entries to the dashboard and then remove all of them.

yusuf-goog replied:
Your correct, right now, the view is going to be messy when we do this. And we should probably find a more graceful way of handling the build view affects in the future (maybe hide flaky by default, or even have a `staging:true` property that hides only the staging ones). 

However, I think utilizing this same mechanism thats used for prod builds to schedule these staging ones are good to do longterm.

keyonghan replied:
That sounds fair. The ideal case if we can easily control what targets to be enabled in prod (show up in dashboard), what targets to to enabled in `bringup: true` (show up in dashboard), and what targets to be enabled in staging for validation purpose only (hide from dashboard), using a single source of truth of the .ci.yaml.

But these need backend refactoring to support. Before we have a clear plan/support, I recommend use existing staging star to validate.

CaseyHillers replied:
I still think the easiest way would be to add `os: Ubuntu` to targets we want to individually migrate, and do it in batches. Once everything is swapped, we can update the platform_properties to default to ubuntu.

yusuf-goog replied:
Switching it that way doesn't allow provide the ability to let them soak for a while. Having the ability to setup staging jobs and let them run as we need would be very helpful. Even if we have to do them in blocks.

yusuf-goog commented:
it would be cool if we could inherit properties, so the staging properties
would only be the delta to the prod properties.

On Tue, May 3, 2022 at 8:15 PM Casey Hillers ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .ci.yaml
> <https://github.com/flutter/flutter/pull/103003#discussion_r864430786>:
>
> > @@ -10,6 +10,23 @@ enabled_branches:
>    - flutter-\d+\.\d+-candidate\.\d+
>
>  platform_properties:
> +  linux_staging_build:
>
> +1 to moving it into the ci.yaml. I think some ways to simplify it is we
> could update the ci.yaml infra to instead not require an explicit staging
> platform_properties, but other than that this approach LGTM
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/flutter/flutter/pull/103003#discussion_r864430786>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AV3Q2C4BR6OLJZFX52GLHHDVIHTUPANCNFSM5VANEOHQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

yusuf-goog committed:
Updating staging build entries to use new naming convention

Staging builds now use a name prefix of Staging_build_, which
allows us to filter these builds on the dashboard.

yusuf-goog committed:
Moving bringup:true to target level

We can't add bringup:true to properties block level, so
need to move this back to each target.

CaseyHillers commented on .ci.yaml:
```diff
@@ -10,6 +10,23 @@ enabled_branches:
   - flutter-\d+\.\d+-candidate\.\d+
 
 platform_properties:
+  staging_build_linux:
```
Synced up offline with Yusuf about this. Using staging_build_linux makes more sense the `Linux_staging` as there's quite a bit of infra and dashboarding that would need to change to support the latter. So doing this is quicker.
